### PR TITLE
Add monkey patch for fenced-frame-src directive

### PR DESCRIPTION
**Pull request by blu25**
This PR introduces a monkey patch that adds a new CSP directive, "fenced-frame-src". This copies most of its behavior from the already existing "frame-src" directive.

***
<a href="https://pr-preview.s3.amazonaws.com/WICG/fenced-frame/pull/34.html" title="Last updated on Jul 17, 2022, 10:20 PM UTC (8b2b847)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/fenced-frame/34/28f6d0c...8b2b847.html" title="Last updated on Jul 17, 2022, 10:20 PM UTC (8b2b847)">Diff</a>

### CODE REVIEW

File: spec.bs [hunk 1]
```diff
@@ -165,3 +169,63 @@ The {{HTMLFencedFrameElement/src}} IDL attribute must [=reflect=] the respective
...
 
 This section details monkeypatches to [[!HTML]]'s <a href="https://html.spec.whatwg.org/multipage/embedded-content-other.html#dimension-attributes">Dimension attributes</a> section. This section will be updated to include <{fencedframe}> in the list of elements that the <code>[=width=]</code> and <code>[=height=]</code> dimension attributes apply to.
+
+<h3 id=new-request-destination>New fenced-frame-src [[!CSP]] [=directive=]</h3>
+
+Fenced frames are a different element from an iframe. Therefore, using the <b><i>[=frame-src=]</i></b> directive wouldn't give web sites enough control over their CSP rules. Introduce a new [[!CSP]] [=directive=]: <b><i>fenced-frame-src</i></b>. The monkey-patched specification is printed below:
+
+<h4 id="directive-fenced-frame-src">`fenced-frame-src`</h4>
+
+The <dfn export>fenced-frame-src</dfn> directive restricts the URLs which may be loaded into
```
Comment (domfarolino) [inline spec.bs:179]: We shouldn't need to export this from _this_ spec, so we can omit `export` for now I think

File: spec.bs [hunk 2]
```diff
@@ -165,3 +169,63 @@ The {{HTMLFencedFrameElement/src}} IDL attribute must [=reflect=] the respective
...
+  directive-value = <a grammar>serialized-source-list</a>
+</pre>
+
+<div id="fenced-frame-src example" class="example">
+  Given a page with the following Content Security Policy:
+  <pre>
+    <a http-header>Content-Security-Policy</a>: <a>fenced-frame-src</a> https://example.com/
+  </pre>
+
+  Fetches for the following code will return a network errors, as the URL
```
Comment (domfarolino) [inline spec.bs:194]: Maybe let's explicitly link to https://fetch.spec.whatwg.org/#concept-network-error?

**Review by domfarolino — CHANGES_REQUESTED** [review]

File: spec.bs [hunk 3]
```diff
@@ -165,3 +169,63 @@ The {{HTMLFencedFrameElement/src}} IDL attribute must [=reflect=] the respective
...
+  provided do not match `fenced-frame-src`'s <a>source list</a>:
+
+  <pre highlight="html">
+    &lt;fencedframe src="https://example.org/"&gt;
+    &lt;/fencedframe&gt;
+  </pre>
+</div>
+
+The <a href="https://w3c.github.io/webappsec-csp/#frame-src-pre-request">Pre-request check</a> and <a href="https://w3c.github.io/webappsec-csp/#frame-src-post-request">Post-request check</a> will be the same as the <a href="https://w3c.github.io/webappsec-csp/#directive-frame-src">frame-src</a>'s check.
```
[the remark this answers is not in the document]
Comment (domfarolino) [reply spec.bs:204]: I think this is fine

File: spec.bs [hunk 4]
```diff
@@ -38,6 +38,8 @@ spec: html; urlPrefix: https://html.spec.whatwg.org/multipage/
     text: height; url: embedded-content-other.html#attr-dim-height
 spec: fetch; urlPrefix: https://fetch.spec.whatwg.org/
   type: dfn
+    text: request; url: #concept-request
```
Comment (domfarolino) [inline spec.bs:41]: This concept is exported by fetch, so we shouldn't need this: https://github.com/whatwg/fetch/blob/main/fetch.bs#L1322

File: spec.bs [hunk 5]
```diff
@@ -38,6 +38,8 @@ spec: html; urlPrefix: https://html.spec.whatwg.org/multipage/
...
+    text: policy; url: #content-security-policy-object
```
Comment (domfarolino) [inline spec.bs:42]: https://fetch.spec.whatwg.org/#content-security-policy-object doesn't seem to link to anything

**Review by domfarolino — CHANGES_REQUESTED** [review]

File: spec.bs (hunk 5 above)
Comment (blu25) [reply spec.bs:42]: I believe this is dead code from when I was trying to put the entirety of the pre and post request checks in the spec.

**Comment by domfarolino** [comment]
Generally looks, good, can you resolve the conflicts and I think we'll be able to get this in!

**Comment by domfarolino** [comment]
Closing and re-opening to see if PR preview is enabled now

**Review by domfarolino — APPROVED** [review]

**Pull request merged**